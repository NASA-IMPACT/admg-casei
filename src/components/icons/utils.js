/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */
import React from "react"

import {
  AtmosphericCompositionIcon,
  // AtmosphericDynamicsIcon,
  CarbonCycleEcosystemsIcon,
  ClimateVariabilityChangeIcon,
  EarthSurfaceInteriorIcon,
  GlobalWaterEnergyCycleIcon,
  WeatherIcon,
  ExclamationIcon,
} from "../icons"

const icons = {
  "df6847ab-9443-4fdb-a990-d5bf1daca3b7": ({ color, size }) => (
    <AtmosphericCompositionIcon color={color} size={size} />
  ),
  "f1bd646f-876f-4e9f-896e-f7d4571ecc00": ({ color, size }) => (
    <CarbonCycleEcosystemsIcon color={color} size={size} />
  ),
  "cd62c68f-a9f7-456a-ae15-53ef79ed5118": ({ color, size }) => (
    <ClimateVariabilityChangeIcon color={color} size={size} />
  ),
  "2231a3f1-3430-4f89-86a2-94938aa000a5": ({ color, size }) => (
    <EarthSurfaceInteriorIcon color={color} size={size} />
  ),
  "fd5c0baa-d6a4-45d7-b201-0eb2a5505dc7": ({ color, size }) => (
    <GlobalWaterEnergyCycleIcon color={color} size={size} />
  ),
  "6d719465-f32c-4808-9efb-da5ad2bae135": ({ color, size }) => (
    <WeatherIcon color={color} size={size} />
  ),
}

export function getFocusIcon(id) {
  return Object.keys(icons).includes(id)
    ? icons[id]
    : ({ color, size }) => <ExclamationIcon color={color} size={size} />
}

export const sizes = {
  text: { width: "16", height: "16" },
  tiny: { width: "30", height: "30" },
  small: { width: "60", height: "60" },
  large: { width: "120", height: "120" },
  huge: { width: "360", height: "360" },
}
