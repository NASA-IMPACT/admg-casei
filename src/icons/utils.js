/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */
import React from "react"

import {
  AtmosphericCompositionIcon,
  // AtmosphericDynamicsIcon,
  CarbonCycleEcosystemsIcon,
  ClimateVariabilityChangeIcon,
  EarthSurfaceInteriorIcon,
  GlobalWaterEnergyCycleIcon,
  WeatherIcon,
  ExclamationIcon,
} from "./"

const icons = {
  "Atmospheric Composition": ({ color, size }) => (
    <AtmosphericCompositionIcon color={color} size={size} />
  ),
  "Carbon Cycle & Ecosystems": ({ color, size }) => (
    <CarbonCycleEcosystemsIcon color={color} size={size} />
  ),
  "Climate Variability & Change": ({ color, size }) => (
    <ClimateVariabilityChangeIcon color={color} size={size} />
  ),
  "Earth Surface & Interior": ({ color, size }) => (
    <EarthSurfaceInteriorIcon color={color} size={size} />
  ),
  "Global Water & Energy Cycle": ({ color, size }) => (
    <GlobalWaterEnergyCycleIcon color={color} size={size} />
  ),
  Weather: ({ color, size }) => <WeatherIcon color={color} size={size} />,
}

export function getFocusIcon(id) {
  return Object.keys(icons).includes(id)
    ? icons[id]
    : ({ color, size }) => <ExclamationIcon color={color} size={size} />
}

export const sizes = {
  text: { width: "16", height: "16" },
  tiny: { width: "30", height: "30" },
  small: { width: "60", height: "60" },
  large: { width: "120", height: "120" },
  huge: { width: "360", height: "360" },
}
