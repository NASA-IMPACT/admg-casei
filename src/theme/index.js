const palette = {
  white: "#ffffff",
  gunmetal: "#262a31",
  "davys grey": "#5f5f5f",
  "dark cornflower blue": "#15418c",
  "rich black fogra 29": "#0c1520",
  "gunmetal 2": "#303641",
  platinum: "#e9eaec",
  "spanish gray": "#9e9e9e",
  lightBlue: "#aac9ff",
  "red ryb": "#f93c21",
  "caribbean green": "#3AD6AB",
  "medium slate blue": "#6F5FE9",
  "orange blossom": "#cb6528",
}

const colors = {
  lightTheme: {
    background: palette.white,
    altBackground: palette.platinum,
    border: "hsla(0, 0%, 0%, 0.2)",
    text: palette.gunmetal,
    altText: palette["davys grey"],
    linkText: palette["dark cornflower blue"],
  },

  darkTheme: {
    background: palette["gunmetal 2"],
    altBackground: palette["rich black fogra 29"],
    text: palette.platinum,
    altText: palette["spanish gray"],
    linkText: palette.lightBlue,
    highlight: palette["red ryb"],
    division: palette["davys grey"],
    dataVizOne: "#1B9E77", // palette["caribbean green"],
    dataVizTwo: "#7570B3", //palette["medium slate blue"],
    dataVizThree: "#D95F02", // palette["orange blossom"],
  },
}

const shape = {
  rounded: "0.2rem",
}

const layout = {
  pageMargin: "5%", //"4rem",
  smallPageMargin: "2rem",
  maxWidth: "1280px",
}

const breakpoints = {
  sm: "800px",
  md: "1280px",
  lg: "1400px",
}

export { colors, shape, layout, breakpoints }
