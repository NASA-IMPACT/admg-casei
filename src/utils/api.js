const fetch = require("node-fetch")

export default {
  async fetchSearchResult(searchstring) {
    try {
      const response = await fetch(
        `http://ec2-3-85-33-46.compute-1.amazonaws.com/api/campaign?search=${searchstring}`,
        {
          method: "GET",
        }
      )
      const json = await response.json()

      if (json.success) {
        return json.data.map(c => c.short_name)
      }
    } catch (error) {
      new Error(error)
    }
  },
}
