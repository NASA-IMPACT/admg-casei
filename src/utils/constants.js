export const POSITIVE = "lightTheme"
export const NEGATIVE = "darkTheme"

export const ALPHABET = [
  "A",
  "B",
  "C",
  "D",
  "E",
  "F",
  "G",
  "H",
  "I",
  "J",
  "K",
  "L",
  "M",
  "N",
  "O",
  "P",
  "Q",
  "R",
  "S",
  "T",
  "U",
  "V",
  "W",
  "X",
  "Y",
  "Z",
]

export const FEEDBACK_FORM_URL = "https://docs.google.com/forms/d/e/1FAIpQLSfCQzZHtaDSiWNdwiJa3TIzsyhjEbYyyHwfrcwIKn_UmdVaKA/viewform"

export const API_DOCUMENTATION_URL = "https://nasa-impact.github.io/admg-backend/documentation/api_doc.html" 