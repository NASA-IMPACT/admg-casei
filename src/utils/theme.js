let color = {
  base: "hsla(0,0%,100%,0.9)", // white
  primary: "#0c1520", // dark background
  secondary: "#303641", // dark blue box background
  tertiary: "#294060", // medium blue for gradiant background
  gray: "#9E9E9E",
  link: "#aac9ff",
  background: "#0c1520",
}

const type = {
  base: {
    root: "16px",
    size: "1rem",
    line: "1.5",
    color: color.base,
    family: '"Titillium Web", sans-serif',
    style: "normal",
    weight: 300,
    light: 300,
    regular: 400,
    medium: 400,
    bold: 700,
  },
  heading: {
    family: '"Titillium Web", sans-serif',
    style: "normal",
    weight: 700,
    light: 300,
    regular: 400,
    medium: 400,
    bold: 700,
  },
}
const shape = {
  rounded: "0.2rem",
  ellipsoid: "320rem",
}
const layout = {
  space: "1rem",
  border: "1px",
  min: "960px",
  max: "1280px",
  maxWidth: 1280,
}
const boxShadow = {
  inset: "inset 0px 0px 3px 0px rgba(0,0,0,0.2);",
  input: "0 -1px 1px 0 rgba(0,0,0,0.08), 0 2px 6px 0 rgba(0,0,0,0.16);",
}
export default {
  layout,
  color,
  type,
  shape,
  boxShadow,
}
/**
 * Media query ranges used by the media utility.
 * They're not exported with the main theme because the utility does not
 * build the media functions in runtime, needing the values beforehand.
 */
export const mediaRanges = {
  xsmall: [null, 543],
  small: [544, 767],
  medium: [768, 991],
  large: [992, 1199],
  xlarge: [1200, null],
}
