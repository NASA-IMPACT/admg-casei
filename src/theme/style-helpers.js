import { css } from "styled-components"

import { colors } from "../theme"

export const scrollFade = css`
  overflow-y: scroll;
  max-height: 10rem;
  display: block;
  margin: 0.5rem;
  background: ${props => `linear-gradient(${
    colors[props.mode].background
  } 33%, rgba(247, 155, 184, 0)),
    linear-gradient(rgba(247, 155, 184, 0), ${
      colors[props.mode].background
    } 66%) 0 100%,
    radial-gradient(
      farthest-side at 50% 0,
      rgba(34, 34, 34, 0.5),
      rgba(0, 0, 0, 0)
    ),
    radial-gradient(
        farthest-side at 50% 100%,
        rgba(34, 34, 34, 0.5),
        rgba(0, 0, 0, 0)
      )
      0 100%`};
  background-color: ${props => colors[props.mode].background};
  background-repeat: no-repeat;
  background-attachment: local, local, scroll, scroll;
  background-size: 100% 45px, 100% 45px, 100% 15px, 100% 15px;
`
export const breakpoints = {
  sm: "800px",
  md: "1280px",
  lg: "1400px",
}
