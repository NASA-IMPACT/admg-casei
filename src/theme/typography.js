import styled from "styled-components"

import theme from "../utils/theme"

export const baseTypeStyle = `
  font-family: "Titillium Web", sans-serif;
  color: ${theme.color.base};
  font-size: 1.25rem;
  font-weight: 400;
`

export const clickableStyle = `
  text-decoration: none;
  cursor: pointer;
  transition: opacity 0.24s ease 0s;

  &:active {
    transform: translate(0 1px);
  }
  &:hover {
    opacity: 0.64;
  }
  &:visited {
    color: ${theme.color.base};
  }
`

export const Heading1 = styled.h1`
  ${baseTypeStyle};
  font-size: 4rem;
  font-weight: 700;
  line-height: 1.25;

  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
`

export const Heading2 = styled.h2`
  ${baseTypeStyle};
  font-size: 3.5rem;
  font-weight: 600;
  line-height: 1.25;

  margin-top: 0.25rem;
  margin-bottom: 1rem;
`

export const Heading3 = styled.h3`
  ${baseTypeStyle};
  font-size: 2.25rem;
  font-weight: 600;
  line-height: 3.75;

  margin-top: 0.25rem;
  margin-bottom: 1rem;
`

export const BodyText = styled.p`
  ${baseTypeStyle};
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.75;

  margin-top: 0;
  margin-bottom: 0;
`

export const SmallBodyText = styled.p`
  ${baseTypeStyle};
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.375;

  margin-top: 0;
  margin-bottom: 0;
`

export const SmallTitle = styled.label`
  ${baseTypeStyle};
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1;

  margin-top: 0;
  margin-bottom: 0;

  text-transform: uppercase;
`

export const Label = styled.label`
  ${baseTypeStyle};
  font-size: 0.75rem;
  font-weight: 600;
  line-height: 1;

  margin-top: 0;
  margin-bottom: 0;
`

export const LinkText = styled.a`
  ${baseTypeStyle};
  font-size: 0.75rem;
  font-weight: 600;
  line-height: 1;

  margin-top: 0;
  margin-bottom: 0;

  ${clickableStyle};
`

export const ButtonText = styled.span`
  ${baseTypeStyle};
  font-size: 0.75rem;
  font-weight: 600;
  line-height: 1;

  margin-top: 0;
  margin-bottom: 0;

  ${clickableStyle};
`

export const Banner = styled.span`
  ${baseTypeStyle};
  font-size: 4rem;
  font-weight: 700;
  line-height: 1.25;

  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
`
