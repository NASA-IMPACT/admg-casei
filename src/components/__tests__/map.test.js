import React from "react"
import renderer from "react-test-renderer"
import mapbox from "mapbox-gl"

import Map from "../map"
import GeoJsonSource from "../map/geojson-source"
import BboxLayer from "../map/bbox-layer"
import { POSITIVE, NEGATIVE } from "../../utils/constants"
import { colors } from "../../theme"

let mapWrapper
let container

describe("Map", () => {
  beforeEach(() => {
    renderer.act(() => {
      mapWrapper = renderer.create(
        <Map height={200}>
          <GeoJsonSource id="test">
            <BboxLayer id="test" bbox={[32, 5, 35, 10]} />
          </GeoJsonSource>
        </Map>,
        {
          createNodeMock: element => {
            container = element
            return element
          },
        }
      )
    })
  })

  it("renders correctly", () => {
    expect(mapWrapper.toJSON()).toMatchSnapshot()
  })

  it("adds a map", () => {
    expect(mapbox.Map).toHaveBeenCalledWith({
      container,
      style: `mapbox://styles/mapbox/satellite-streets-v11/`,
      zoom: 1,
      center: [0, 0],
    })
  })
})

const sourceId = "test-source"
const layerId = "test-layer"
const geojson = {
  type: "FeatureCollection",
  features: [],
}
const bbox = [32, 5, 35, 10]
let sourceWrapper, layerWrapper

describe("GeoJsonSource", () => {
  it("adds a map source", () => {
    renderer.act(() => {
      sourceWrapper = renderer.create(
        <GeoJsonSource id={sourceId} map={mapbox.Map()} geojson={geojson} />
      )
    })

    expect(sourceWrapper.toTree().props.map.addSource).toHaveBeenCalledWith(
      `${sourceId}-source`,
      {
        type: "geojson",
        data: geojson,
      }
    )
  })
})

describe("BboxLayer", () => {
  beforeEach(() => {
    renderer.act(() => {
      layerWrapper = renderer.create(
        <BboxLayer
          id={layerId}
          bbox={bbox}
          sourceId={sourceId}
          map={mapbox.Map()}
        />
      )
    })
  })
  it("adds a map layer", () => {
    expect(layerWrapper.toTree().props.map.addLayer).toHaveBeenCalledWith({
      id: `${layerId}-layer`,
      layout: {},
      paint: {
        "line-color": colors[NEGATIVE].text,
        "line-opacity": 0.8,
        "line-width": 2,
      },
      source: sourceId,
      type: "line",
    })
  })

  //ensure switch is working as expected with "campaign", "explore" or "test"
  it("does not zoom to the data", () => {
    expect(layerWrapper.toTree().props.map.flyTo).not.toHaveBeenCalled()
  })

  describe("when id is 'campaign'", () => {
    it("zooms to the data", () => {
      renderer.act(() => {
        layerWrapper = renderer.create(
          <BboxLayer
            id="campaign"
            bbox={bbox}
            sourceId={sourceId}
            map={mapbox.Map()}
          />
        )
      })

      expect(layerWrapper.toTree().props.map.flyTo).toHaveBeenCalledWith({
        center: [-73.5804, 45.53483],
        pitch: 60,
        bearing: -60,
        zoom: 10,
      })
    })
  })

  describe("when id is 'explore'", () => {
    it("zooms to the data", () => {
      renderer.act(() => {
        layerWrapper = renderer.create(
          <BboxLayer
            id="explore"
            bbox={bbox}
            sourceId={sourceId}
            map={mapbox.Map()}
          />
        )
      })

      expect(layerWrapper.toTree().props.map.flyTo).toHaveBeenCalledWith({
        center: [-73.5804, 45.53483],
        pitch: 60,
        bearing: -60,
        zoom: 10,
      })
    })
  })
})
