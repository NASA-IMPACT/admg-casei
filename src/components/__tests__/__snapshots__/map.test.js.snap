// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders correctly 1`] = `
<div
  className="map___StyledDiv-sc-j9lea3-0 iORfhW"
  data-cy="mapboxgl-map"
/>
`;
